jupiter-tests/src/test/java/org/junit/jupiter/engine/discovery/DiscoveryTests.java

```diff
@@ -44,9 +44,13 @@
 import org.junit.jupiter.engine.descriptor.ClassTestDescriptor;
 import org.junit.jupiter.engine.descriptor.NestedClassTestDescriptor;
 import org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor;
+import org.junit.jupiter.engine.kotlin.KotlinDefaultImplsTestCase;
+import org.junit.jupiter.engine.kotlin.KotlinInterfaceImplementationTestCase;
+import org.junit.jupiter.engine.kotlin.KotlinInterfaceTestCase;
```

Comment:
I'm pretty sure this will fail in Eclipse.

So, I'll likely have to revise these tests to use reflection once it's on `main`.

Reply:
We do something similar in `TestInstanceLifecycleKotlinTests`. Does that work in Eclipse?

Reply:
If you mean the following, yes that "works".

https://github.com/junit-team/junit-framework/blob/9a6b0a7110396ad04fe342723f5ed9c8a235df8d/jupiter-tests/jupiter-tests.gradle.kts#L51-L55

Reply:
But I was actually thinking about something like the following, so that the rest of the tests still run in Eclipse.

https://github.com/junit-team/junit-framework/blob/9a6b0a7110396ad04fe342723f5ed9c8a235df8d/junit-vintage-engine/src/test/java/org/junit/vintage/engine/VintageTestEngineExecutionTests.java#L867-L872